tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/awaitility/poc/AsyncClientTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.awaitility.poc;
+
+import org.assertj.core.api.Assertions;
+import org.awaitility.core.ConditionTimeoutException;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+
+import static org.awaitility.Awaitility.await;
+import static org.awaitility.Awaitility.with;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertEquals;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.testng.Assert.fail;
+
+
+import java.io.File;
+//import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Future;
+import java.util.regex.Pattern;
+
+public class AsyncClientTest  extends Arquillian {
+
+    @Inject
+    private AsyncClient client;
+
+    @Deployment
+    public static WebArchive deploy() {
+        File[] libs = Maven.resolver().loadPomFromFile("pom.xml")
+            .resolve("org.awaitility:awaitility", "org.assertj:assertj-core").withTransitivity()
```

Comment:
You could create an Arquillian extension that would add Awaitility to all tested archives, similarly to e.g. this: https://github.com/rhoar-qe/thorntail-test-suite/blob/master/common/src/main/java/org/wildfly/swarm/ts/common/AssertjArquillianExtension.java

Reply:
Thanks, @Ladicek for share this info, ok I will include the extension with the refactoring that I'm currently working on.

Reply:
The extension was added and is working now with the asynchronous tests. Thank you